### PR TITLE
ENH: Add pick_types and crop

### LINKED ISSUE #1138: ENH/API: add cropping  / tmin-tmax paramters to time_frequency functions to avoid edge artifacts
... like fieldtrip does.  Also related to GSOC-topic-4

### LINKED ISSUE #1270: ENH: add pick_types instance method to DropPickChannels Mixin
... avoids writing code like:

```
keep_ch = [epochs.ch_names[i] for in pick_types(epochs.info, meg=True)]
epochs_meg = epochs.pick_channels(keep_ch, copy=True)
```

cc @adykstra @agramfort

### PR DESCRIPTION
**Pull request by Eric89GXL**
Closes #1138.
Closes #1270.

Ready for review/merge.

### CODE REVIEW

**Comment by agramfort** [comment]
Lgtm

Should we deprecate pick types evoked?

I think the pick types method should appear in some examples

Also what's new

Thx

**Comment by Eric89GXL** [comment]
@agramfort done

**Comment by Eric89GXL** [comment]
@agramfort you want me to merge once AppVeyor comes back happy?

**Comment by agramfort** [comment]
LGTM

File: examples/inverse/plot_gamma_map_inverse.py [hunk 1]
```diff
@@ -57,10 +57,10 @@
 
 # Show the evoked response and the residual for gradiometers
 ylim = dict(grad=[-120, 120])
-evoked = mne.pick_types_evoked(evoked, meg='grad', exclude='bads')
+evoked.pick_types(meg='grad', exclude='bads')
 evoked.plot(titles=dict(grad='Evoked Response Gradiometers'), ylim=ylim,
             proj=True)
 
-residual = mne.pick_types_evoked(residual, meg='grad', exclude='bads')
+residual.pick_types(meg='grad', exclude='bads')
```
Comment (dengemann) [inline examples/inverse/plot_gamma_map_inverse.py]: @Eric89GXL why don't we add this everywhere. Would be lovely as an addition to pick/drop_channels. I would suggest to add this to the related mixin.

**Pull request merged**